pest/examples/parens.rs

```diff
@@ -73,3 +77,18 @@ fn main() {
         };
     }
 }
+
+#[cfg(feature = "miette")]
+fn main() -> miette::Result<()> {
+    loop {
+        let mut line = String::new();
+
+        print!("> ");
+        io::stdout().flush().into_diagnostic()?;
+
+        io::stdin().read_line(&mut line).into_diagnostic()?;
+        line.pop();
+
+        ParenParser::parse(Rule::expr, &line)?;
+    }
+}
```

Comment:
This would be better off keeping the behavior of `loop { try { ... } }`. That would be the same as the other `main`, e.g.

```suggestion
fn main() {
    loop {
        let mut line = String::new();

        print!("> ");
        io::stdout().flush().into_diagnostic()?;

        io::stdin().read_line(&mut line).unwrap();
        line.pop();

        let parse = ParenParser::parse(Rule::expr, &line);
        #[cfg(feature = "miette")]
        let parse = parse.map_err(<miette::Report as From>::from);
        match parse {
            Ok(pairs) => println!("{:?}", expr(pairs)),
            Err(e) => println!("\n{}", e),
        }
    }
}
```

(untested)

Reply:
Unfortunately, it seems `Report`'s display only displays a message:

```
> x

Failed to parse at (1, 1)
>
```

Without any colors, or fancy source code formatting. The real fix would probably be to use an `impl ReportHandler`, but in order to see that, I need to acquire a `&mut Formatter`...

Reply:
You can use `{:?}` here to get the return-error-from-main formatting.

To get `&mut Formatter`, you `impl Display::fmt`.